torch/_dynamo/variables/dicts.py

```diff
@@ -429,6 +430,116 @@ def var_getattr(self, tx, name: str) -> "VariableTracker":
+    # when is_matching_cls(cls) is true
+    @classmethod
+    def create(cls, user_cls, args, kwargs, options):
+        CustomizedDictVariable._patch_once(user_cls)
+
+        if not args and not kwargs:
+            # CustomDict() init with empty arguments
+            raw_items = collections.OrderedDict()
+        elif dataclasses.is_dataclass(user_cls):
+            # @dataclass CustomDict(a=1, b=2)
+            bound = inspect.signature(user_cls).bind(*args, **kwargs)
+            bound.apply_defaults()
+            raw_items = bound.arguments
+        elif len(args) == 1 and isinstance(args[0], ConstDictVariable) and not kwargs:
+            # CustomDict({'a': 1, 'b': 2})
+            raw_items = args[0].items
+        else:
+            unimplemented("custome dict init with args/kwargs unimplemented")
+
+        items = collections.OrderedDict()
+        for key in raw_items.keys():
+            val = raw_items[key]
+            if isinstance(val, VariableTracker):
+                items[key] = val
+            elif variables.ConstantVariable.is_literal(val):
+                items[key] = variables.ConstantVariable(val)
+            else:
+                unimplemented("expect VariableTracker or ConstantVariable.is_literal")
+
+        return cls(items, user_cls, **options)
+
+    # called from builder.py
+    @classmethod
+    def wrap(cls, builder, obj):
+        raise NotImplementedError()
+
+    def __init__(self, items, user_cls, **options):
+        super().__init__(items, user_cls, **options)
+        assert self.is_matching_cls(user_cls)
+
+    def as_proxy(self):
+        raise NotImplementedError()
+
+    # 'RETURN_VALUE triggered compile'
+    # called from torch/_dynamo/codegen.py
+    def reconstruct(self, codegen):
+        codegen.extend_output([codegen._create_load_const(self.user_cls)])
+        keys = tuple(self.items.keys())
+        for key in keys:
+            codegen(self.items[key])
+        return codegen.create_call_function_kw(len(keys), keys, True)
+
+    def call_method(
+        self,
+        tx,
+        name,
+        args: "List[VariableTracker]",
+        kwargs: "Dict[str, VariableTracker]",
+    ) -> "VariableTracker":
+        options = VariableTracker.propagate(self, args, kwargs.values())
+        fn = getattr(self.user_cls, name)
+        source = None if self.source is None else AttrSource(self.source, name)
+
+        if hasattr(fn, "__objclass__") and fn.__objclass__ in (
+            dict,
+            collections.OrderedDict,
+        ):
+            # for python dict method without overridden
+            return super().call_method(tx, name, args, kwargs)
+        elif name in ("__getitem__", "to_tuple", "__setitem__", "__setattr__"):
+            # for user overridden method
+            return tx.inline_user_function_return(
+                variables.UserFunctionVariable(fn, source=source, **options),
+                [self] + list(args),
+                kwargs,
+            )
```

Comment:
I think for ```__setitem__``` and ```__setattr__```, we probably can't simple inlining them, especially if the attribute to be set is a function. cc @jansel